soc/silabs/common/soc_siwx91x_power_pmgr.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024-2025 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/pm.h>
+#include "sl_si91x_power_manager.h"
+#include "sli_si91x_clock_manager.h"
+#include "sl_rsi_utility.h"
+#include "sl_si91x_m4_ps.h"
+
+LOG_MODULE_REGISTER(siwx91x_pm);
+
+extern uint32_t frontend_switch_control;
+
+void sli_si91x_m4_ta_wakeup_configurations(void);
+
+/*
+ * Power state map:
+ * PM_STATE_RUNTIME_IDLE: SL_SI91X_POWER_MANAGER_STANDBY (PS4)
+ * PM_STATE_SUSPEND_TO_IDLE: SL_SI91X_POWER_MANAGER_SLEEP (PS4 Sleep)
+ */
+void pm_state_set(enum pm_state state, uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	uint32_t target_power_state = 0;
+
+	switch (state) {
+	case PM_STATE_RUNTIME_IDLE:
+		target_power_state = SL_SI91X_POWER_MANAGER_STANDBY;
+		break;
+
+	case PM_STATE_SUSPEND_TO_IDLE:
+		target_power_state = SL_SI91X_POWER_MANAGER_SLEEP;
+		break;
+
+	default:
+		/* Unsupported power state */
+		LOG_ERR("Unsupported power state %d", state);
+		irq_unlock(0);
+		return;
+	}
+
+	/* Set PRIMASK */
+	__disable_irq();
+	/* Set BASEPRI to 0. */
+	irq_unlock(0);
+
+	if (sl_si91x_power_manager_is_ok_to_sleep()) {
+		/* Device is ready to sleep; proceed with sleep operations. */
+		if (target_power_state == SL_SI91X_POWER_MANAGER_STANDBY) {
+
+			sl_si91x_power_manager_standby();
+		} else {
+			if (sli_si91x_config_clocks_to_mhz_rc() != 0) {
+				LOG_ERR("Failed to configure clocks for sleep mode");
+			} else {
+#if SL_WIFI_COMPONENT_INCLUDED
```

Comment:
You probably want to rely on a Zephyr symbol (`CONFIG_WISECONNECT_NETWORK_STACK`?).